pyccel/codegen/printing/ccode.py

```diff
@@ -1921,14 +1899,18 @@ def _print_Constant(self, expr):
         if expr == math_constants['inf']:
             self.add_import(c_imports['math'])
             return 'HUGE_VAL'
+        elif expr == math_constants['nan']:
+            self.add_import(c_imports['math'])
+            return 'NAN'
         elif expr == math_constants['pi']:
             self.add_import(c_imports['math'])
             return 'M_PI'
         elif expr == math_constants['e']:
             self.add_import(c_imports['math'])
             return 'M_E'
         else:
-            raise NotImplementedError("Constant not implemented")
+            return self._print(expr.value)
```

Comment:
Shouldn't this look as above `self._print(LiteralFloat(expr.value))`? To ensure the proper precision?

Reply:
was not sure if we have a complex type constant for example would that still work ?

Reply:
Hmm, good point I hadn't thought of that. Does `Constant` have a dtype? I still think the precision is necessary

Reply:
```
    >>> from pyccel.ast.variable import Constant
    >>> import math
    >>> Constant('float', 'pi' , value=math.pi )
    Constant('pi', dtype=NativeFloat())
```

this is the example in the Class so i think yes we can pass the precision because the value is a `Variable`.

Reply:
By "I still think the precision is necessary", I meant that we need to print `2.33_F64`, but I realise that this doesn't necessarily apply to C. I don't know if the precision is ever anything other than the default?

If we have the type though then we can do:
```python
elif expr.dtype is NativeFloat():
    return self._print(LiteralFloat(expr.value))
elif expr.dtype is NativeComplex():
    return self._print(LiteralComplex(expr.value))
....
```

Reply:
is there a way to use the class and pyccelize it ?
i am trying to create some tests

I agree that we should at least use the types.

the precision can be specified when creating the Constant
`Constant('float', 'pi' , precision=8,value=math.pi )`
other ways it is the default.

Reply:
i still need to change the Fortran printer as it print all as `LiteralFloat`

Reply:
> is there a way to use the class and pyccelize it ?
i am trying to create some tests

See `tests/ast/test_basic.py`

Reply:
> the precision can be specified when creating the Constant
Constant('float', 'pi' , precision=8,value=math.pi )
other ways it is the default.

If the precision is stored in the `Constant` anyway, then you can directly use `DtypePrecisionToCastFunction`